mypy/plugins/functools.py

```diff
@@ -229,9 +229,11 @@ def partial_new_callback(ctx: mypy.plugin.FunctionContext) -> Type:
             partial_names.append(fn_type.arg_names[i])
         elif actuals:
             if any(actual_arg_kinds[j] == ArgKind.ARG_POS for j in actuals):
+                # Don't add params for arguments passed positionally
                 continue
+            # Add defaulted params for arguments passed via keyword
             kind = actual_arg_kinds[actuals[0]]
-            if kind == ArgKind.ARG_NAMED:
+            if kind == ArgKind.ARG_NAMED or kind == ArgKind.ARG_STAR2:
```

Comment:
I know this works at runtime, but is this a common pattern that we should allow? I mean do people often use `partial(f, x=1)(x=2)` etc?

Reply:
I have at least one such use in a checked codebase.